src/ui/public/state_management/config_provider.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * @name stateManagementConfig
+ *
+ * @description Allows apps to configure state management
+ */
+
+import { uiModules } from 'ui/modules';
+
+uiModules.get('kibana/state_management')
+  .provider('stateManagementConfig', class StateManagementConfigProvider {
+    enabled = true
```

Comment:
__nit:__ `enabled` is public and writable hence we implicitly declare two alternative ways to change it: directly and via `enable/disable` methods. Can we at least mark it as "private" with `_` prefix?